include/deepworks/base_op.hpp

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include <deepworks/placeholder.hpp>
+#include <deepworks/call.hpp>
+#include <deepworks/layer_info.hpp>
+
+namespace deepworks {
+
+template <typename R, class C, typename ... Types>
+constexpr size_t get_num_ins( R(C::*)(Types ...)) {
```

Comment:
I think get_number_of_arguments is more clear

Reply:
and it's much longer as well :)

Reply:
get_num_args is Ok ?

Reply:
Done